buildkite/resource_pipeline_team_test.go

```diff
@@ -0,0 +1,226 @@
+package buildkite
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+// Confirm we can add and remove a team pipeline resource with the default access level
+func TestAccPipelineTeam_AddRemoveWithDefaultsAccess(t *testing.T) {
+	var tp pipelineTeamResourceModel
+	teamName := acctest.RandString(12)
+	t.Parallel()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: protoV6ProviderFactories(),
+		CheckDestroy:             testCheckPipelineTeamResourceRemoved,
+		Steps: []resource.TestStep{
+			{
+				Config:             testAccPipelineTeamConfigBasic(teamName, "READ_ONLY"),
+				ExpectNonEmptyPlan: true,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Confirm the test resource team & team pipeline exists in the Buildkite API
+					testAccCheckPipelineTeamExists("buildkite_pipeline_team.pipelineteam", &tp),
+					// Confirm the team pipeline has the correct values in Buildkite's system
+					testAccCheckPipelineTeamRemoteValues("READ_ONLY", &tp),
+					// Confirm the team pipeline has the correct values in terraform state
+					resource.TestCheckResourceAttrSet("buildkite_pipeline_team.pipelineteam", "team_id"),
+					resource.TestCheckResourceAttrSet("buildkite_pipeline_team.pipelineteam", "pipeline_id"),
+					resource.TestCheckResourceAttr("buildkite_pipeline_team.pipelineteam", "access_level", "READ_ONLY"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccPipelineTeam_AddRemoveWithNonDefaultAccess(t *testing.T) {
+	var tp pipelineTeamResourceModel
+	teamName := acctest.RandString(12)
+	t.Parallel()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: protoV6ProviderFactories(),
+		CheckDestroy:             testCheckPipelineTeamResourceRemoved,
+		Steps: []resource.TestStep{
+			{
+				Config:             testAccPipelineTeamConfigBasic(teamName, "BUILD_AND_READ"),
+				ExpectNonEmptyPlan: true,
```

Comment:
Same with this. I'd expect an empty plan here

Reply:
same as above reason for adding the ExpectNonEmptyPlan